src/genworld_gui.cpp

```diff
@@ -350,9 +378,17 @@ struct GenerateLandscapeWindow : public Window {
 				}
 				break;
 
+			case WID_GL_TOWNNAME_DROPDOWN: {
+				uint gen = _settings_newgame.game_creation.town_name;
+				StringID name = gen < BUILTIN_TOWNNAME_GENERATOR_COUNT ?
+						STR_GAME_OPTIONS_TOWN_NAME_ORIGINAL_ENGLISH + gen :
+						GetGRFTownNameName(gen - BUILTIN_TOWNNAME_GENERATOR_COUNT);
```

Comment:
pedantry: not sure i agree with the line splitting here. All on one line, or split with the `?`/`:` as the first character